Applications/Spire/Include/Spire/Ui/RealSpinBox.hpp

```diff
@@ -98,7 +98,8 @@ namespace Spire {
       QString display_string(Real value);
       QStyle::SubControl get_current_control(const QPoint& mouse_pos);
       boost::optional<Real> get_value(const QString& text) const;
-      bool is_valid(const QString& string);
+      bool is_valid_input(const QString& string);
+      bool is_valid_value(const QString& string);
```

Comment:
What is the difference been a value value and a valid input?

Reply:
is_valid_input(...) checks if the input text itself has the *potential* to make a valid number (i.e., accepts 'partial' input), but also prevents things like entering a negative sign when the minimum is >= 0.

is_valid_value(...) is the final check to see if the input text makes a valid value within the bounds of the SpinBoxModel.

Reply:
To answer the question more directly: a valid value is valid with respect to the model, and a valid input is valid with respect to the UX, but may or may not be acceptable as a submitted value.